src/proto/Oasis.proto

```diff
@@ -0,0 +1,38 @@
+// Copyright © 2017-2020 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+syntax = "proto3";
+
+package TW.Oasis.Proto;
+option java_package = "wallet.core.jni.proto";
+
+message TransferMessage {
+    string to = 1;
+    string method = 2;
+    uint64 gas_price = 3;
+
+    // Amount values strings prefixed by zero. e.g. "\u000010000000"
+    bytes gas_amount = 4;
+    // Amount values strings prefixed by zero
+    bytes amount = 5;
```

Comment:
can we change it to string? we plan to move to string for big int for new coins

Reply:
Because of the way Oasis indicates the sign of the amount, we need to transport amount's as bytes, so we can represent the sign.

Reply:
is it possible to set negative value? after reading this comment: https://github.com/trustwallet/wallet-core/pull/1258/files#r565283797 I suggest to hide this detail implementation side c++ code, so that android/ios/sdk users can just set simple string value and don't need to care about this sign thingy

if `string` is not enough to express an amount, we can turn it to a message like this (from Cosmos.proto): 

```proto
message Amount {
    string denom = 1;
    int64 amount = 2;
}
```

Reply:
@emmanuelm41 can you guys solve this one?

Reply:
I am actually trying to convert to string this field and test it. We think if we use that Amount definition you suggested to us the tas will not work on oasis network.

Reply:
why not? this is more friendly to the client, you still can convert string to types inside cpp code

Reply:
Lots of classes are created automatically from the .proto definition file (java classes, cpp classes). If we want to hide the bytes type and change it to string, it requires to create a different proto definition and then translate it somehow somewhere. We made a deep analysis and we think it's really hard to make that happen.

Reply:
please check `Elrond.proto` value field and see how it handles

Reply:
What do you want me to see in that coin? I checked it, and the value field is string type everywhere. It goes like that in the proto file, in all the code. Am I missing something?

Reply:
just change type in proto! and handle the string conversion inside your cpp code (from string to bytes), that would make client much easier to handle, because it's a weird encoding doesn't have to expose to outside, not that hard to change

Reply:
But about Elrond, as fas as I can check, it always uses that value as string everywhere, and never has to manipulate it into bytes before encoding. The network requires the amount and gas amount to be bytes. The serialization into CBOR is made by autogenerated proto but code. If I change it into string, i will need to create two definitions. One with this value as string and one with this value as bytes. Uses the string definition internally, and before serialize the tx to CBOR, copy all the values to a new message with this other proto definition, and then serialize it. It's a really complex change to do. I am sorry if I don't get what you are suggesting to me, but I really think there is no easy way to do it.

Reply:
when it's called from android:
```kotlin
 amount = ByteString.copyFrom("10000000".toByteArray()) // decimal 10000000
```
from swift:
```swfit
$0.amount = Data("10000000".utf8)
```

string is straightforward, `amount = "10000000"`

Reply:
I know that! I already made that change to test it. That's easy to do. But...

**CBOR tx with string type on value**
82880000620030406900313030303030303000707374616b696e672e5472616e7366657278636f617369732d636f72652f636f6e73656e7375733a20747820666f7220636861696e2062633163373135333139313332333035373935666138366264333265393332393161616163626662356235393535663362613738626462613431336166396531584055bc9e3394baf149ad16df8a1fb7815f8ec11113a76589a9c54e6668f88ed547b163943b96019a09a1ba534279e6ddb48500e443d473e9ad97468bd63247540b

**CBOR tx with bytes type on value**
82880000420030404900313030303030303000707374616b696e672e5472616e7366657278636f617369732d636f72652f636f6e73656e7375733a20747820666f7220636861696e20626331633731353331393133323330353739356661383662643332653933323931616161636266623562353935356633626137386264626134313361663965315840cfe091da8d2d148a053591f435d711adf6f79105a73343bdfe32a30520c5a421d32429a52eb2d5d70ef3d161712abf3696d4072d4289c9771a6e78f5c0299d0b

If you put that output on [http://cbor.me](url), you will be able to see the differences. The type goes in CBOR tx. The oasis node check the each field types. As you can see here
[https://github.com/Zondax/ledger-oasis/blob/eb60f0ae781270d2a28505f4abad64a1fbf8a478/app/src/consumer/parser_impl_con.c#L160](url)

So, at last we cannot send it as string when we encode the data on CBOR. The code uses the auto generated types from proto definition. I should have two proto definitions. One to encode data at last, and one to create and handle the tx through the code (cpp, Kotlin, swift)

Reply:
Encoding in the proto and in the encoded tx (CBOR) are different things!  Make it string in the proto, parse to numerical in Signer before encoding.  Proto format is only used between swift/kotlin client and C++ lib, it does not affect encoded format.

Reply:
I'm confused too, you meant you had tried to encode string into COBR message here? https://github.com/trustwallet/wallet-core/pull/1258/files#diff-3b8c8445c8d509bebc4dc6b053c2cf559920bd7be423775418b43417aa8789baR33? should it be converted to bytes here?

Reply:
Thanks a lot to both of you. The misunderstood was te relationship between CBOR encoding and .proto file. I made the changes, and I think the tests will still be green.